staging/src/k8s.io/client-go/plugin/pkg/client/auth/oidc/oidc.go

```diff
@@ -59,9 +61,13 @@ const expiryDelta = 10 * time.Second
 
 var cache = newClientCache()
 
-// Like TLS transports, keep a cache of OIDC clients indexed by issuer URL.
+// Like TLS transports, keep a cache of OIDC clients indexed by issuer URL. This ensures
+// current requests from different clients don't concurrently attempt to refresh the same
+// set of credentials.
 type clientCache struct {
-	mu    sync.RWMutex
+	mu sync.RWMutex
+	// Each oidcAuthProvider holds a mutex to ensure a set of credentials only refresh
+	// their token once.
```

Comment:
does this assume short-lived processes? refreshing a token multiple times could be valid